homeassistant/components/spc/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config flow for SPC integration."""
+import logging
+
+from pyspcwebgw import SpcWebGateway
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import CONN_CLASS_LOCAL_PUSH, ConfigFlow
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+
+from . import CONF_API_URL, CONF_WS_URL, DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_API_URL, description={"suggested_value": "http://<IP>:8088"}
+        ): str,
+        vol.Required(
+            CONF_WS_URL, description={"suggested_value": "ws://<IP>:8088/ws/spc"}
+        ): str,
+    }
+)
+
+
+class SpcConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a SPC config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
```

Comment:
This `try/except` feels too big. Focus on just the logic that could throw one of these exceptions.

Reply:
Definitely. Fixed!